### PR TITLE
Report voted fractions for a given root

### PR DESCRIPTION
**Pull request by potuz**
This PR exposes a forkchoice function to report the fraction of the committee that voted for a given root. 

This is just an approximation, it considers the active balance of the node divided by the total active balance in the epoch divided by the number of slots per epoch. It returns it as a percentage value. 

Notice that when requesting historical roots in forkchoice, the values may be slightly off since the store only tracks the latest active validator balance.

### CODE REVIEW

File: beacon-chain/forkchoice/doubly-linked-tree/types.go [hunk 1]
```diff
@@ -40,6 +40,7 @@ type Store struct {
 	highestReceivedSlot           types.Slot                            // The highest received slot in the chain.
 	receivedBlocksLastEpoch       [fieldparams.SlotsPerEpoch]types.Slot // Using `highestReceivedSlot`. The slot of blocks received in the last epoch.
 	allTipsAreInvalid             bool                                  // tracks if all tips are not viable for head
+	committeeBalance              uint64                                // tracks the average validator balance per slot: requires a lock on nodes to read/write
```
Comment (terencechain) [inline beacon-chain/forkchoice/doubly-linked-tree/types.go:43]: I don't think this comment is right. Do you mean average committee balance? I wish my validator as that balance...

Comment (potuz) [reply beacon-chain/forkchoice/doubly-linked-tree/types.go:43]: ah yeah, it's the average total active validators per slot, will update in a sec

File: beacon-chain/forkchoice/doubly-linked-tree/node_test.go [hunk 2]
```diff
@@ -261,3 +261,45 @@ func TestNode_SetFullyValidated(t *testing.T) {
...
+	vote, err = f.VotedFraction([32]byte{'b'})
+	require.NoError(t, err)
+	require.Equal(t, uint64(0), vote)
+
+	// After we call head the voted fraction is obtained.
+	_, err = f.Head(context.Background(), balances)
+	require.NoError(t, err)
+	vote, err = f.VotedFraction([32]byte{'b'})
+	require.NoError(t, err)
+	require.Equal(t, uint64(1), vote)
```
Comment (terencechain) [inline beacon-chain/forkchoice/doubly-linked-tree/node_test.go:300]: can we add a test for a value that's not `0` or `1`? like `0.5`

File: beacon-chain/forkchoice/doubly-linked-tree/unrealized_justification.go [hunk 3]
```diff
@@ -90,12 +90,14 @@ func (s *Store) pullTips(state state.BeaconState, node *Node, jc, fc *ethpb.Chec
...
 	}
 
-	uj, uf, err := precompute.UnrealizedCheckpoints(state)
+	ab, uj, uf, err := precompute.UnrealizedCheckpoints(state)
 	if err != nil {
 		log.WithError(err).Debug("could not compute unrealized checkpoints")
 		uj, uf = jc, fc
 	}
 
+	s.committeeBalance = ab / uint64(params.BeaconConfig().SlotsPerEpoch)
```
[the remark this answers is not in the document]
Comment (potuz) [reply beacon-chain/forkchoice/doubly-linked-tree/unrealized_justification.go:99]: yes

File: beacon-chain/forkchoice/doubly-linked-tree/node_test.go (hunk 2 above)
Comment (potuz) [reply beacon-chain/forkchoice/doubly-linked-tree/node_test.go:300]: 1 is 1%, not 100

**Review by terencechain — APPROVED** [review]

**Pull request merged**